### PR TITLE
Revise deployment script

### LINKED ISSUE #30: [NFT] - deploy script
* add `base_uri` to Matty constructor
* read `base_uri` from `state.${network}.json` and deploy contract in `deploy` script

### PR DESCRIPTION
**Pull request by robertu7**
* revise contracts and deployment script;
  * set `name_`, `symbol_`, `supply_` and `sharedBaseURI_` on constructor while deploying;
* move logbook test cases to `Logbook.test.ts`;
* estimate gas used by preOrder functions;

### CODE REVIEW

File: contracts/BatchNFT.sol [hunk 1]
```diff
@@ -13,26 +13,25 @@ contract BatchNFT is ERC721, Ownable {
     using Counters for Counters.Counter;
     Counters.Counter internal _tokenIds;
 
-    uint16 public _totalSupply;
+    uint16 public totalSupply;
```
Comment (robertu7) [inline contracts/BatchNFT.sol:16]: updated [naming convention](https://paper.dropbox.com/doc/Smart-Contract--BUaGChU3o36bcXzcdDj5k76GAQ-GIDkfXqyQsqjmAc1IRt6f):

- input variables ends with `_`: `lowerCamelCase_`
- internal/private variables starts with `_`: `_lowerCamelCase`
- public variables: `lowerCamelcase`

File: contracts/BatchNFT.sol [hunk 2]
```diff
@@ -13,26 +13,25 @@ contract BatchNFT is ERC721, Ownable {
...
+        setSharedBaseURI(sharedBaseURI_);
     }
 
-    receive() external payable {}
-
     /**
      * @dev Withdraw all ether in the contract
      */
-    function withdrawAll(address vault_) public onlyOwner {
+    function withdrawAll(address vault_) public payable onlyOwner {
```
Comment (robertu7) [inline contracts/BatchNFT.sol:34]: added `payable` and removed `receive() external payable {}`

Comment (guoliu) [reply contracts/BatchNFT.sol:34]: Curious why this function needs to be payable?

**Review by guoliu — APPROVED** [review]
🌱

File: contracts/BatchNFT.sol (hunk 2 above)
Comment (robertu7) [reply contracts/BatchNFT.sol:34]: just for [testing](https://github.com/thematters/matters-avatar/pull/37/files#diff-ee4c354a2a46ddb89a4678af8d6e4749fefefd658910e922fe70c2ff2f965b64R84), not in use in production.

**Pull request merged**